src/Kernel/client/telemetry.ts

```diff
@@ -34,94 +36,97 @@ class LiteEvent<T> implements ILiteEvent<T> {
-            var script = document.createElement('script');
+        return new Promise((resolve) => {
+            const script = document.createElement('script');
             script.onload = () => { resolve(); };
             script.type = 'text/javascript';
             script.src = javascriptUrl;
             document.head.append(script);
         });
     }
 
-    public async requestConsent(consentMarkup: ConsentMarkup) {
-        console.log(`Adding Cookie API Stylesheets to the document: ${consentMarkup.Stylesheets.length} files`);
-        await Promise.all(consentMarkup.Stylesheets.map((css, _, __) => this.addStylesheet(css)));
-        console.log(`Cookie API Stylesheets loaded.`);
+    private onConsentChanged(categoryPreferences: Record<consentCategories, boolean>) {
+        console.log("onConsentChanged", categoryPreferences);
+        this.checkRequiredConsent();
+    }
+
+    private checkRequiredConsent() {
+        const hasConsent = this.siteConsent.getConsentFor(consentCategories.Required);
+        console.log(`HasConsent: ${hasConsent}`);
+        if (hasConsent) {
+            this.onConsentGranted();
+        }
+    }
 
-        var div = document.createElement('div');
-        div.innerHTML = consentMarkup.Markup;
+    public async requestConsent() {
+        const div = document.createElement('div');
+        div.id = "cookie-banner";
         document.body.prepend(div);
-        console.log("Cookie API markup div added");
+        console.log("Cookie banner div added");
 
-        console.log(`Adding Cookie API Javascripts to the document: ${consentMarkup.Javascripts.length} files`);
-        await Promise.all(consentMarkup.Javascripts.map((javascript, _, __) => this.addJavascript(javascript)));
+        console.log(`Adding Cookie API Javascript to the document`);
+        await this.addJavascript(COOKIE_CONSENT_JS);
         console.log(`Cookie API Javascripts loaded.`);
 
-        this.mscc = mscc;
-        if (this.mscc == null) {
+        if (WcpConsent === null) {
             console.log("Cookie API javascript was not loaded correctly. Consent cannot be obtained.");
             return;
         }
 
-        console.log("Calling Cookie API hasConsent");
-        this.mscc.on('consent', this.onConsentGranted);
-        var hasConsent = this.mscc.hasConsent();
-        console.log(`HasConsent: ${hasConsent}`);
-        if (hasConsent) {
-            this.onConsentGranted();
-        }
+        console.log("Initializing WcpConsent...");
+        const userLanguage = navigator.language;
+        WcpConsent && WcpConsent.init(userLanguage, div.id, (err, siteConsent) => {
+            if (!err) {
+                this.siteConsent = siteConsent;
+                console.log("WcpConsent initialized.");
+                this.checkRequiredConsent();
+            } else {
+                console.log("Error initializing WcpConsent: " + err);
+            }
+        }, this.onConsentChanged, Themes.light);
```

Comment:
Should we detect if the user is in high-contrast mode and, if so, pass the appropriate theme here for accessibility? This may be outside the scope of this change, but it would be good to create an issue so we don't forget about it.

Reply:
The theme parameter is a new cookie API parameter to have the banner to better blend in the webpage look & feel. For that we would need to understand/support Jupyter themes work and that's out of scope.
From the accessibility standpoint, the banner will display correctly, matching the OS high-contrast settings (I just tested on Mac and Windows).